lib/biker.rb

```diff
@@ -1,3 +1,25 @@
 class Biker
+  attr_reader :name, :max_distance, :rides, :acceptable_terrain
+
+  def initialize(name, max_distance)
+    @name = name
+    @max_distance = max_distance
+    @rides = {}
+    @acceptable_terrain = []
+  end
+
+  def learn_terrain!(terrain)
+    @acceptable_terrain << terrain unless @acceptable_terrain.include?(terrain)
+  end
+
+  def log_ride(ride, time)
+    @rides[ride] ||= []
```

Comment:
This is a nice option! It's also totally fine to just have a simple if statement in here to verify if the key exists or not, and add the time appropriately:
```
if @rides.has_key?(ride)
    @rides[ride] << time
else
    @rides[ride] = [time]
end
```